### ISSUE TITLE
Construct your own training set and test set

### ISSUE THREAD

**Issue by dfx1822375** [timeseriesAI/tsai#452]
I would like to ask about loading my own data and constructing training and test sets. The examples I viewed were all UCR datasets, but how do I construct a dataset by reading my own csv or mat file? What are the specific functions to use? For example, I have stored 200 (sequence length) x 3 (send channel) x 30 (receive channel) x 7500 (number of samples) in csv, corresponding to 7500 tags, each tag is a number from 1 to 150 representing the type of sample, how can I use tsai to construct a training test set?

**Comment by oguiza** [comment]
Hi @dfx1822375,
`tsai` can handle inputs with shape [n_samples x n_vars x seq length]. The data you talk about has 4 dimensions instead of 3. I'm not sure what the difference between receive channel and send channel. So you would need to reshape your input data to meet the input entry criteria (maybe merging send and receive channels into 1 dimension?). That would constitute the X. The tag would be the y.

**Comment by oguiza** [comment]
Hi @dfx1822375, 
There are some examples on how to prepare your data in the [documention](https://timeseriesai.github.io/tsai/data.preparation.html). I'm not sure if they fit your needs since I don't fully understand what's your starting point.

**Comment by dfx1822375** [comment]
> Hi @dfx1822375, There are some examples on how to prepare your data in the [documention](https://timeseriesai.github.io/tsai/data.preparation.html). I'm not sure if they fit your needs since I don't fully understand what's your starting point.

Can I get your contact information to communicate? Thank you very much!

**Comment by oguiza** [comment]
> Can I get your contact information to communicate? Thank you very much!
timeseriesAI@gmail.com
